src/Serializer/ContributorNormalizer.php

```diff
@@ -114,24 +117,12 @@ public function normalize($contributor, $format = null, array $context = []): ar
             'preview' => $this->getImageAbsoluteUrl($contributor, 'previewImageFile'),
             'contributions' => $contributor->getNoticesCount(),
             'contribution' => [
-                'example' => $exampleNotice && $exampleNoticeMatchingContexts ? [/* Deprecated */
-                    'matchingUrl' => $exampleNoticeMatchingContexts->first() ? $exampleNoticeMatchingContexts->first()->getExampleUrl() : null,
-                    'noticeId' => $exampleNotice->getId(),
-                    'noticeUrl' => $this->noticeUrlGenerator->generate($exampleNotice),
-                    'screenshot' => $this->getImageAbsoluteUrl($exampleNotice, 'screenshotFile'),
-                ] : null,
-                'starred' => $exampleNotice && $exampleNoticeMatchingContexts ? [
-                    'matchingUrl' => $exampleNoticeMatchingContexts->first() ? $exampleNoticeMatchingContexts->first()->getExampleUrl() : null,
-                    'noticeId' => $exampleNotice->getId(),
-                    'noticeUrl' => $this->noticeUrlGenerator->generate($exampleNotice),
-                    'screenshot' => $this->getImageAbsoluteUrl($exampleNotice, 'screenshotFile'),
-                ] : null,
+                'example' => $starred, /* Deprecated */
+                'starred' => $starred, /* Deprecated */
                 'pinnedNotices' => $pinnedNotices->map(function (Notice $notice) {
-                    $matchingContexts = $notice->getMatchingContexts();
-
                     return [
                         'sort' => $notice->getPinnedSort(),
-                        'matchingUrl' => $matchingContexts && $matchingContexts->first() ? $matchingContexts->first()->getExampleUrl() : null,
+                        'matchingUrl' => $notice->getMatchingUrl(),
```

Comment:
Nice refactor ! Previous code looks ugly now :-)

I’m just not sure about the new entity method name … It seems to me that `getExampleUrl()' describes best : I want to have an example of an URLS where this notice can be seen. `getMatchingUrl()` makes me think that there is only one … It could also be `getExampleMatchingUrl()' but seems a bit long for nothing

No big deal anyway

Reply:
I don't mind either, let's go for `getExampleMatchingUrl` there's never enough descriptive code :p
> https://github.com/dis-moi/extension/pull/730 on this too